athenz/assertion.go

```diff
@@ -90,11 +171,88 @@ func resourceAssertionSchema() *schema.Schema {
 					Optional: true,
 					Default:  false,
 				},
+				"id": {
+					Type:     schema.TypeInt,
+					Computed: true,
+				},
+				"condition": {
+					Type:     schema.TypeSet,
+					MaxItems: 2, // each assertion represent acl policy. Since in a given service,
+					// you can make the acl policy enforced on some hosts and not on others,
+					// therefor, to apply more than 2 conditions per assertion doesn't make sense
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"id": {
+								Computed: true,
+								Type:     schema.TypeInt,
+							},
+							"instances": {
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem:     resourceConditionSchema(nil),
+							},
+							"enforcementstate": {
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem: resourceConditionSchema(
+									validation.ToDiagFunc(
+										validation.StringInSlice([]string{"report", "enforce"}, false)),
+								),
+							},
+							"scopeonprem": {
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem:     resourceConditionSchema(validateDiagFuncForScope),
+							},
+							"scopeaws": {
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem:     resourceConditionSchema(validateDiagFuncForScope),
+							},
+							"scopeall": {
+								Type:     schema.TypeSet,
+								Required: true,
+								MaxItems: 1,
+								Elem:     resourceConditionSchema(validateDiagFuncForScope),
```

Comment:
maybe I am missing something - but why these fields are set with only 1 item?

Reply:
If you look on AssertionCondition class definition you see that the value of each key in the assertion conditionsMap is single AssertionConditionData.